runtime/port/unix/j9sysinfo.c

```diff
@@ -1628,6 +1652,23 @@ j9sysinfo_get_cache_info(struct J9PortLibrary *portLibrary, const J9CacheInfoQue
 	int32_t result = J9PORT_ERROR_SYSINFO_NOT_SUPPORTED;
 	Trc_PRT_sysinfo_get_cache_info_enter(query->cmd, query->cpu, query->level, query->cacheType);
 
+	/* We need to avoid checking the cache directory as the cache info doesn't exist
+	 * on Linux booted via QEMU (the emulator). In addition, it is still unclear
+	 * whether all RISC-V CPUs support the cache line in this way. If no cache info
+	 * is available on Linux, we should not assume it is literally supported on the hardware.
+	 * In such case, there is no padding adjustment for class objects
+	 * even if the -XX:-RestrictContended option is specified on the command line.
+	 */
+#if defined(RISCV64)
+	DIR* cacheDir = opendir("/sys/devices/system/cpu/cpu0/cache");
+	if (NULL != cacheDir) {
+		closedir(cacheDir);
+	} else {
+		Trc_PRT_sysinfo_get_cache_info_exit(result);
+		return result;
+	}
```

Comment:
This isn't legal C code (the declaration of `cacheDir` follows the call to `Trc_PRT_sysinfo_get_cache_info_enter()` above): nest these 7 lines in a block to fix that.
The comment belongs within the `#if defined(RISCV64)` section as well.

Reply:
@keithc-ca ,  the code there has been fixed against your comment.

In addition, I checked the result cache directory on the HiFive U540 Dev board (booting Fedora_riscv) and the results are as follows:
```
[root@openj9riscv ~]# ls  -l /sys/devices/system/cpu/cpu0/cache/index0
total 0
-r--r--r--. 1 root root 4096 Mar 30 20:02 coherency_line_size
-r--r--r--. 1 root root 4096 Dec 15 05:47 level
-r--r--r--. 1 root root 4096 Mar 30 20:10 number_of_sets
-r--r--r--. 1 root root 4096 Mar 30 20:10 shared_cpu_list
-r--r--r--. 1 root root 4096 Mar 30 20:10 shared_cpu_map
-r--r--r--. 1 root root 4096 Dec 15 05:47 size
-r--r--r--. 1 root root 4096 Dec 15 05:47 type
-rw-r--r--. 1 root root 4096 Mar 30 20:10 uevent
-r--r--r--. 1 root root 4096 Mar 30 20:10 ways_of_associativity

[root@openj9riscv ~]# cat  /sys/devices/system/cpu/cpu0/cache/index0/level
1
[root@openj9riscv ~]# cat  /sys/devices/system/cpu/cpu0/cache/index0/coherency_line_size
64  <------
[root@openj9riscv ~]# cat  /sys/devices/system/cpu/cpu0/cache/index1/level
1
[root@openj9riscv ~]# cat  /sys/devices/system/cpu/cpu0/cache/index1/coherency_line_size
64 <------

[jincheng_riscv@openj9riscv ~]$ getconf LEVEL1_DCACHE_LINESIZE
0 <------ (no data out there)
```
Based on results above, it indicates:
1) these cache related directories do exist when the Linux image is booted on the hardware. 2) there is no support for `sysconf `at least on Fedora_riscv.

After talking with the Fedora/RISCV developer at #fedora-riscv/IRC as to `sysconf `,  I was told this function can be can added to Fedora/RISCV but this is not in their wishlist (not priority to do so) for now and there is no plan when this function is supported on Fedora/RISCV.

Under such circumstance, we have to stick to check the cache related directory as it is more reliable as compared to `sysconf ` on the hardware.